File: 4. Java/EX30.java
package Apostila;

import java.util.Scanner;

public class EX30
{
    public static void main(String[] args) 
    {
        // Inicializa variáveis e contadores
        int cod = 0, alc = 0, gas = 0, die = 0;

        // Scanner para entrada de dados
        try (Scanner ent = new Scanner(System.in))
        {
            // Loop até que o código seja 4
            while (cod != 4) 
            {
                // Solicita o código
                System.out.println("# Informe um codigo (1, 2, 3) ou 4 para parar: ");
                cod = ent.nextInt();

                // Processa o código com switch
                switch (cod)
                {
                    case 1:
                        alc++; // Alcool
                        break;

                    case 2:
                        gas++; // Gasolina
                        break;

                    case 3:
                        die++; // Diesel
                        break;

                    case 4:
                        System.out.println("# Encerrando o programa");
                        break;

                    // Mensagem para opções inválidas
                    default:
                        System.out.println("# Opcao invalida!\n");
                }
            }
        }

        // Exibe os totais finais fora do loop
        System.out.println("# MUITO OBRIGADO");
        System.out.println("# Alcool: " + alc);
        System.out.println("# Gasolina: " + gas);
        System.out.println("# Diesel: " + die);
    }
}

File: 6. Python/EX30.py
# Inicializa contadores
cod = 0
gas = 0
die = 0
alc = 0

# Executa enquanto o código for diferente de 4
while cod != 4:
    # Solicita um código ao usuário
    cod = int(input("# Informe um código (1, 2, 3) ou 4 para parar: "))

    # Dicionário com funções para incrementar contadores
    switch = {
        1: lambda: globals().__setitem__("alc", alc + 1),  # Incrementa álcool
        2: lambda: globals().__setitem__("gas", gas + 1),  # Incrementa gasolina
        3: lambda: globals().__setitem__("die", die + 1)   # Incrementa diesel
    }

    # Verifica e executa o incremento conforme o código
    if cod in switch:
        switch[cod]()

# Exibe a mensagem de agradecimento e os resultados
print("\n# MUITO OBRIGADO")
print(f"# Álcool: {alc}")
print(f"# Gasolina: {gas}")
print(f"# Diesel: {die}")
